### DISCUSSION TITLE
Sprites with own fonts

### DISCUSSION THREAD

**Discussion by error105** [Bodmer/TFT_eSPI#1919] [General]
Hi, i have problem with my own fonts :(
I convert my ttf font on 2 sites:
1. https://rop.nl/truetype2gfx/
2. https://tomeko.net/online_tools/file_to_hex.php?lang=en
But when i paste in code: gaugeBack.loadFont(OwnFont); it wont display, or screen is fully black.

I have in code "const uint8_t  OwnFont[] PROGMEM = {};"

**Comment by Bodmer** [comment]
loadFont is for smooth (anti-aliased) fonts only not GFX format fonts.

Use this example a a reference sketcxh:

https://github.com/Bodmer/TFT_eSPI/tree/master/examples/Generic/Local_Custom_Fonts

**Reply by error105** [reply]
When i use: gaugeBack.setFreeFont(&mr2font); and then gaugeBack.drawNumber(obroty, 120, 180, 1); i have error output:
no matching function for call to 'TFT_eSprite::setFreeFont(const uint8_t (*)[924])'

**Reply by error105** [reply]
void createBackground() {
  gaugeBack.setColorDepth(8);
  gaugeBack.createSprite(240, 240);
  gaugeBack.setPivot(120, 120);
  tft.setPivot(160, 120);
  gaugeBack.fillSprite(TFT_TRANSPARENT);
  gaugeBack.pushImage(0, 0, 240, 240, SpeedoWHT);
  gaugeBack.setTextColor(COLORS);
  gaugeBack.setTextDatum(MC_DATUM);
  gaugeBack.setFreeFont(mr2font);
  gaugeBack.drawNumber(obroty, 120, 180, GFXFF);
  }

**Comment by error105** [comment]
void createBackground() {
  gaugeBack.setColorDepth(8);
  gaugeBack.createSprite(240, 240);
  gaugeBack.setPivot(120, 120);
  tft.setPivot(160, 120);
  gaugeBack.fillSprite(TFT_TRANSPARENT);
  gaugeBack.pushImage(0, 0, 240, 240, SpeedoWHT);
  gaugeBack.setTextColor(COLORS);
  gaugeBack.setTextDatum(MC_DATUM);
  gaugeBack.setFreeFont(mr2font);
  gaugeBack.drawNumber(obroty, 120, 180, GFXFF);
  }
